_posts/2019-04-05-omakase-typescript.markdown

```diff
@@ -0,0 +1,171 @@
+specifically to make very, very large amounts of easy to read with - which means sometimes you have to do more
+writing/reading chores. It's an implicit vs explicit thing.
+
+### What is TypeScript?
+
+TypeScript is a programming language - it contains all of JavaScript, and then a bit more. So in our case above,
+lets compare the scripts for "Hello, Danger" in JavaScript vs TypeScript:
+
+```js
+// JavaScript
+var name = "Danger"
+console.log("Hello, " + danger)
+
+// TypeScript
+var name = "Danger"
+console.log("Hello, " + danger)
+
+// Yep, you're not missing something, there's no difference
+```
+
+Due to TypeScript aiming to only _extend_ JavaScript, your normal JavaScript code should work fine with TypeScript.
+The parts of the JavaScript language which TypeScript extends are specifically about trying to make it easier to be
+more explicit in your code.
+
+```diff
+- var name = "Danger"
++ var name: string = "Danger"
+console.log("Hello, " + danger)
+```
+
+This allows the reader to be certain that `name` will only be a string. Annotating your variables also allows
+TypeScript to check this for you.
+
+Simply speaking, we call these annotation "Types" and this is the "Type" in TypeScript. The tag-line for TypeScript
+is "JavaScript which scales" which is a statement that these extra type annotations allows you to work on bigger
+projects. This is because you can verify how correct your code is, and as many people read and write it then you
+don't need to fit it all in your head.
+
+TypeScript extends JavaScript more, but this is the core concept - and that's enough for now.
+
+### Why does Artsy use TypeScript?
+
+Artsy definitely isn't the size of Microsoft! Artsy is about 30 engineers, and Microsoft are about 60,000. However,
+some of our problems are the same. The "front end" developers at Artsy build apps which are thousands of files big.
+Any single change to one file, could affect any other file. A good way to think about this is how in a story,
+changing a conversation earlier can trigger a butterfly effect of changes later down the line.
+
+Typically, the need to ensure all your changes do what you want is less of a problem for people building websites.
+Websites are easy to make changes to, because if you change the site - everyone gets the update instantly. However,
+we also build our iOS app with JavaScript, and a change to the app requires Apple to review the changes and for
+someone to download the new version from the App Store.
+
+This means that the iOS team needs to have more checks that everything is OK before shipping the app to the world.
+Using TypeScript gives our team the ability to feel good that the changes we have made are only the changes we
+want.
+
+TypeScript isn't the only programming language to tackle the problem of making JavaScript code safer, but it's the
+one with the biggest community, allows people to re-use their JavaScript knowledge and has a really good
+programmers text editor.
```

Comment:
This sentence should be split up. Might be more impactful to talk about 'extra tools to help developers work faster' than 'text editor' specifically.